tests/PromoCodesTest.php

```diff
@@ -0,0 +1,206 @@
+
+        $this->assertEquals($promoCode, end($promoCodesAfterCreate));
+    }
+
+    public function testCannotCreateDuplicatePromoCode()
+    {
+        $promoCodeCode = 'TEST-' . time();
+        $promoCode = [
+            'code' => $promoCodeCode,
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ];
+        $addedPromoCode = $this->client->createPromoCode($this->testMaintainerId, $promoCode);
+
+        $this->assertEquals($promoCode['code'], $addedPromoCode['code']);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(400);
+        $this->expectExceptionMessage(sprintf('Promo code %s already exists', $promoCodeCode));
+
+        $this->client->createPromoCode($this->testMaintainerId, $promoCode);
+    }
+
+    public function testOrganizationAdminCannotListPromoCode()
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(403);
+
+        $this->normalUserClient->listPromoCodes($this->testMaintainerId);
+    }
+
+    public function testOrganizationAdminCannotCreatePromoCode()
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(403);
+        $this->normalUserClient->createPromoCode($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ]);
+    }
+
+    public function testRandomAdminCannotCreatePromoCode()
+    {
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(403);
+        $this->normalUserClient->createPromoCode($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ]);
+    }
+
+    public function testInvalidOrganization()
+    {
+        try {
+            $this->client->createPromoCode($this->testMaintainerId, [
+                'code' => 'TEST-' . time(),
+                'expirationDays' => rand(5, 20),
+                'organizationId' => 0,
+                'projectTemplateStringId' => 'poc6months',
+            ]);
+            $this->fail('Organization not found');
+        } catch (ClientException $e) {
+            $this->assertEquals(404, $e->getCode());
+        }
+    }
+
+    public function testInvalidProjectTemplate()
+    {
+        $organization = $this->client->createOrganization($this->testMaintainerId, [
```

Comment:
Spis bych to ale zkousel testovat pod _ normalUserem_ a jako `projectTemplateStringId` pouzil kod ProjectTemplatesTest::TEST_HIDDEN_PROJECT_TEMPLATE_STRING_ID